.pre-commit-config.yaml

```diff
@@ -0,0 +1,39 @@
+fail_fast: true
+
+exclude: |
+  (?x)^(
+    .*/(snapshots)/.*|
+  )$
+
+repos:
+  - repo: https://github.com/abravalheri/validate-pyproject
+    rev: v0.16
+    hooks:
+      - id: validate-pyproject
```

Comment:
Do we need this one?

Reply:
I think Ruff has support for this?

Reply:
Validate-pyproject validates schemas. Ruff just makes sure the pyproject is parsable, as far as I know, and checks its own section. It doesn't check schemas (yet?).

It's not that useful though unless you add the schemas from SchemaStore, which can be done either with --store (slow, pulls from network) or `validate-pyproject-schema-store`, which is a regularly updated copy of the pyproject related schemas. Currently about half of the most popular projects are supported with schemas: https://github.com/SchemaStore/schemastore/issues/3564.

Reply:
Yeah, this was literally just copied from Ruff :D